sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -4117,8 +4172,15 @@ def apply_fn(df):
   @property  # type: ignore
   @frame_base.with_docs_from(DataFrameGroupBy)
   def dtypes(self):
-    grouping_columns = self._grouping_columns
-    return self.apply(lambda df: df.drop(grouping_columns, axis=1).dtypes)
+    return frame_base.DeferredFrame.wrap(
+        expressions.ComputedExpression(
+            'dtypes',
+            lambda gb: gb.dtypes,
+            [self._expr],
+            requires_partition_by=partitionings.Arbitrary(),
+            preserves_partition_by=partitionings.Arbitrary()
+        )
+    )
```

Comment:
It turns out the old implementation was relying on incorrect behavior in `apply`, so I've updated this not to use `apply`

Reply:
Nice